src/Idea.js

```diff
@@ -13,17 +13,28 @@ const style = {
 
 
 
-export const Idea = ({id,left,top,hideSourceOnDrag,children, }) => {
+export const Idea = ({id,left,top,hideSourceOnDrag,children,displayEdit }) => {
     const [{ isDragging }, drag] = useDrag({
         item: { id, left, top, type: ItemTypes.IDEA},
         collect: (monitor) => ({
             isDragging: monitor.isDragging(),
         }),
     });
+    
+    
+
+    const handleClick = useCallback((e) => {
+        e.preventDefault();
+        displayEdit(id);
+    }, [displayEdit, id]);
+
+
+
+
     if(isDragging && hideSourceOnDrag) {
         return <div ref={drag}/>;
     }
-    return (<div ref={drag} style={{...style, left, top}}>
+    return (<div ref={drag} style={{...style, left, top}} onDoubleClick={handleClick}>
         {children}
     </div>);
```

Comment:
```suggestion
    return (
        <div ref={drag} style={{...style, left, top}} onDoubleClick={handleClick}>
            {children}
        </div>
    );
```

You should always line up HTML open and close tags so that it's easier to read.